functions/Set-DbaStartupParameter.ps1

```diff
@@ -207,43 +206,48 @@ function Set-DbaStartupParameter {
             }
         } else {
             Write-Message -Level Verbose -Message "Offline switch set, proceeding with just WMI"
-            $Server = $SqlInstance
+            $server = $SqlInstance
         }
 
-        #Get Current parameters:
-        $currentstartup = Get-DbaStartupParameter -SqlInstance $SqlInstance -Credential $Credential
-        $originalparamstring = $currentstartup.ParameterString
+        # Get Current parameters (uses WMI) -- requires elevated session
+        try {
+            $currentStartup = Get-DbaStartupParameter -SqlInstance $SqlInstance -Credential $Credential -EnableException
+        } catch {
+            Stop-Function -Message "Unable to gather current startup parameters" -Target $SqlInstance -ErrorRecord $_
+            return
+        }
```

Comment:
This is particularly important to do anytime we are calling other public functions in our commands. Because this was not including `EnableException` and wrapped in a try/catch it was falsely assuming SingleUser was true at current line 237 because the object didn't have any content to begin with.... so `f (!($currentStartup.SingleUser))` will return true when that property is true *AND* when `$currentStartup -eq $null` ...so the object itself had no content.